pkg/controller/controllercmd/builder.go

```diff
@@ -64,12 +89,38 @@ func (b *ControllerBuilder) WithInstanceIdentity(identity string) *ControllerBui
 }
 
 // Run starts your controller for you.  It uses leader election if you asked, otherwise it directly calls you
-func (b *ControllerBuilder) Run() error {
+func (b *ControllerBuilder) Run(stopCh <-chan struct{}) error {
 	clientConfig, err := b.getClientConfig()
 	if err != nil {
 		return err
 	}
 
+	if b.servingInfo != nil {
+		kubeConfig := ""
+		if b.kubeAPIServerConfigFile != nil {
+			kubeConfig = *b.kubeAPIServerConfigFile
+		}
+		serverConfig, err := serving.ToServerConfig(*b.servingInfo, *b.authenticationConfig, *b.authorizationConfig, kubeConfig)
+		if err != nil {
+			return err
+		}
+		serverConfig.HealthzChecks = append(serverConfig.HealthzChecks, b.healthChecks...)
+
+		server, err := serverConfig.Complete(nil).New(b.componentName, genericapiserver.NewEmptyDelegate())
+		if err != nil {
+			return err
+		}
+
+		go func() {
+			server.PrepareRun().Run(stopCh)
+			glog.Fatal("server exited")
+		}()
+	} else {
```

Comment:
else if

Reply:
even better: check the error case `len(b.healthChecks) > 0 && b.servingInfo == nil` first.